kafka-client/src/main/java/dev/responsive/kafka/internal/clients/ResponsiveRestoreConsumer.java

```diff
@@ -101,7 +106,23 @@ public ConsumerRecords<K, V> poll(final Duration timeout) {
       log.error("Found uninitialized changelog partitions during poll: {}", uninitializedOffsets);
       throw new IllegalStateException("Restore consumer invoked poll without initializing offsets");
     }
-    return super.poll(timeout);
+
+    try {
+      return super.poll(timeout);
+    } catch (final OffsetOutOfRangeException e) {
+      // e.partitions() should never be empty, but we check it because accidentally
+      // passing in empty partitions is catastrophic (it will seek all partitions to
+      // beginning)
+      if (repairRestoreOffsetOutOfRange && !e.partitions().isEmpty()) {
+        log.warn("The restore consumer attempted to seek to offsets that are no longer in range. "
```

Comment:
if we log `offsetOutOfRangePartitions()` we'll get the fetched offsets in the log too